### ISSUE TITLE
Is it possible to spy Context?

### ISSUE THREAD

**Issue by sdelaysam** [linkedin/dexmaker#59]
Hi,

Whether it is possible to do the following?
```
Context context = spy(InstrumentationRegistry.getTargetContext());
```

It fails on emulator API 17 (however `mock()` works without a problem)
```
W/dalvikvm: Superclass of 'LContextImpl_Proxy;' (Landroid/app/ContextImpl;) is not accessible
W/dalvikvm: Link of class 'LContextImpl_Proxy;' failed
```

Here is my build.gradle
```
androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
        exclude group: 'com.android.support', module: 'support-annotations'
})
androidTestCompile 'com.linkedin.dexmaker:dexmaker:2.2.0'
androidTestCompile 'com.linkedin.dexmaker:dexmaker-mockito:2.2.0' 
```

What I've already tried:
1) Add file `androidTest/resources/mockito-extension/org.mockito.plugins.MockMaker` with the following contents
```
com.android.dx.mockito.DexmakerMockMaker
```
2)  Add this line to `setup()` method
```
System.setProperty("dexmaker.dexcache", getInstrumentation().getTargetContext().getCacheDir().getPath());
```

And still no luck... 
Any advice, guys?

**Comment by moltmann** [comment]
This should work when using dexmaker-mockito-inline (which requires a current aosp/master or eventually a Android P build).

**Comment by drewhannay** [comment]
@sdelaysam I released version 2.16.0 today, which is the first release that includes the new `dexmaker-mockito-inline` artifact. Please try it out and let us know if you have any issues!

Keep in mind, as @moltmann mentioned, this new artifact can only be used if you're running your tests on Android P (which is currently in Developer Preview 1)
